### PR TITLE
Fix ima measurement logic & upload file before failure

### PR DESCRIPTION
**Pull request by lilyeyes**
Fix ima measurement logic & upload file before failure
Failed case can be tracked by bug:
  https://bugzilla.suse.com/show_bug.cgi?id=1172223

- Related ticket: https://progress.opensuse.org/issues/67243
- Needles: NA
- Verification run: http://openqa.suse.de/tests/4291033#

### CODE REVIEW

**Comment by rfan1** [comment]
Only one comment here, it is better with more clear abstract 
e.g. Fix ima measurement logic & upload file before failure :)

**Comment by lilyeyes** [comment]
> Only one comment here, it is better with more clear abstract
> e.g. Fix ima measurement logic & upload file before failure :)

done

**Comment by jouyingbin** [comment]
> > Only one comment here, it is better with more clear abstract
> > e.g. Fix ima measurement logic & upload file before failure :)
> 
> done

Hi lily, just wondering do you also update the git commit title too ?

**Comment by lilyeyes** [comment]
yes, but there are character lenth limitation (50)

File: tests/security/ima/ima_measurement.pm [hunk 1]
```diff
@@ -40,11 +40,15 @@ sub run {
     $self->wait_boot;
     $self->select_serial_terminal;
 
+    # Upload files for reference before test dies
```
Comment (lemon-suse) [inline tests/security/ima/ima_measurement.pm:43]: Please update the comments here for the reason why need upload file before test die.

**Comment by lilyeyes** [comment]
I think it is very obvious here. e.g,
if test fails in code of "line 48", then code of "line 60 upload" will not happen.

**Comment by lemon-suse** [comment]
> I think it is very obvious here. e.g,
> if test fails in code of "line 48", then code of "line 60 upload" will not happen.

It will waste time to upload the file even it is passed, so it is better to use post_hook.

**Comment by lilyeyes** [comment]
> > I think it is very obvious here. e.g,
> > if test fails in code of "line 48", then code of "line 60 upload" will not happen.
> 
> It will waste time to upload the file even it is passed, so it is better to use post_hook.
Agree with you, will enhance all the test modules later.
But atm I would like to keep this code for referance as IMA changes a lot recently.

**Comment by lemon-suse** [comment]
LGTM.

**Pull request merged**